### ISSUE TITLE
Shorter default Notes

### ISSUE THREAD

**Issue by Lolligerhans** [glinscott/fishtest#596]
When creating a new test and the "Notes" textbox is left empty, it is filled with the whole commit message of the tip. That's pretty cool. :) I suggest using some kind of separator ("` | `", "` -- `", "`: `") other than "` `". This would suit the typical subject-body structure of commits.

**Comment by Lolligerhans** [comment]
Clarification: The separator is meant to be inserted after the first line of the commit message, if there are multiple.

**Comment by Lolligerhans** [comment]
Alternatively remove all lines but the first (hence the issue title).

I found see few people using this feature. I think it's very helpfull, so some tweaks to avoid 

- very long notes or
- breaking the separation of subject and body

may make it more popular.

**Comment by vondele** [comment]
@Lolligerhans yes, I think we should encourage people to write good commit messages also for their tests on fishtest. Maybe we make progress at a later stage to add search to fishtest and be able to search the commit messages to return a list of tests (see https://github.com/glinscott/fishtest/issues/339). If I understand your issue correctly you suggest to display commit messages a little differently. 

* Replace the first two newlines of a commit message by a visual hint (e.g. '--') so that the title stands out. (for those commit messages that are multi line). 
* truncate the remainder of the lines to e.g. 140 characters so that on the main page the comments box is not overly full (or such that the comment fits 3 lines, depending on the window width).  

What one could imagine is to display on the main page the truncated comment as above, while the full fishtest comment is shown on the tests/view/... page.

**Comment by vondele** [comment]
actually the 'info' field is already on the tests/view page... so we're just talking about truncation.

A test like https://tests.stockfishchess.org/tests/view/5e9690efc2718dee3c822961 has about 193 char in the info field... that's still pretty reasonable, but could be roughly the threshold for truncation ? @Lolligerhans that matches your expectation?

**Comment by Lolligerhans** [comment]
@vondele Yes, these were exactly my concerns. Keeping the full message on the test page and only truncate for the overview sounds like a good idea! Maybe show an ellipsis ("`...`") if truncation takes place.

A conservative 150 to 200 chars may be a good start. Most knotty are Situation where I try to split screen on my single display.

I try to use the subject line of a first test to tell the other testers what my change/idea roughly is about in case they want to improve/comment on it. I use the body for hints to the approver and notes for my future self. At followups I only use the subject line for a take number and the detail I tweaked.
Not sure if that's the best way, but generally, the overview simply has no need for a longer text to me. The typical 50 char subject line tends to be very short, but 2 to 4 lines are just fine. That would allow to reserve the commit message subject line for a take number (as most do), and still have some info shown in the overview.

So to summarize my proposition would be [`/tests`]:

1. Replace the first "`\n\n`" by "` | `" (splice subject line with separator).
2. Truncate from the _second_ occurence of "`\n\n`" (allows to deliberately hide text which is not really relevant).
3. If 150 characters are exceeded, truncate at the next whitespace (or "`. `" or "`\n`", whatever you prefer).
4. Replace remaining "`\n`" with "`  `" (some > 1 amount of space).
5. If 200 characters are exceeded, truncate the rest.
6. Whenever truncation occurs,indicate with an ellipsis.

and [`/tests/view/...`]:

1. Do not truncate.
2. Splice subject like above.
3. Replace newlines like above (maybe replace  "`\n\+`" with "`\n`" if you prefer).

**Comment by Lolligerhans** [comment]
I assume `^[Bb]ench[: \t] [0-9]+\n` can always be removed pretty safely.

**Comment by Lolligerhans** [comment]
When writing someting in the box, would also be great to have a checkbox that says "append default" and appends the generated default text to the typed message.

For those writing messages like "start LTC if I fall asleep". ;)
